### PR TITLE
Add missing X-Pack repositories REST specs

### PR DESCRIPTION
**Pull request by stevejgordon**
While working on the .NET client code-gen, I noticed that a new experimental X-Pack API was added for repositories in 7.10.0, but is missing a REST spec required for client code generation. This PR adds the specs for the two new endpoints.

API was added in #60371

Please let me know what (if anything) needs to happen to backport to 7.10 and 7.x?

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/api/repositories.clear_metering_archive.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "repositories.clear_metering_archive":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/clear-repositories-metering-archive-api.html",
+      "description":"Removes the archived repositories metering information present in the cluster."
+    },
+    "stability":"experimental",
+    "url":{
+      "paths":[
+        {
+          "path":"/_nodes/{node_id}/_repositories_metering/{max_version_to_clear}",
```
Comment (Mpdreamz) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/repositories.clear_metering_archive.json:11]:
```suggestion
          "path":"/_nodes/{node_id}/_repositories_metering/{version}",
```

**Review by Mpdreamz — APPROVED** [review]
LGTM

**Comment by Mpdreamz** [comment]
@fcofdez this LGTM, mind giving this a quick look over since this is your new API? 👍

**Comment by Mpdreamz** [comment]
Jenkins run elasticsearch-ci/2 please

File: x-pack/plugin/src/test/resources/rest-api-spec/api/repositories.clear_metering_archive.json (hunk 1 above)
Comment (fcofdez) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/repositories.clear_metering_archive.json:11]: I'll prefer to leave that parameter name, as it clears up to the specified version.

**Review by fcofdez — APPROVED** [review]
LGTM

**Comment by stevejgordon** [comment]
Apologies @fcofdez, I've just noticed that I let this PR get stale. Since these still need to be added I have rebased them and updated them to include some of the latest spec fields such as `visibility`. Which version labels would be best to add these? Would 7.15.0 and 8.0.0 be enough?

**Comment by fcofdez** [comment]
No worries @stevejgordon :). `7.15.0` and `8.0.0` versions are fine. 👍

**Pull request merged**